core/EE_Registry.core.php

```diff
@@ -1354,7 +1354,8 @@ protected function _resolve_dependencies(ReflectionClass $reflector, $class_name
                     $arguments,
                     $index
                 );
-            } else {
+            }
+            if (empty($arguments[ $index ])) {
```

Comment:
Is this location the best location for this conditional check?  Is it something that could happen (and would be more efficient) earlier in the loop iteration?

Reply:
no because it is represents the fallback behaviour when a dependency can not be resolved any other way and therefore has to happen after all other attempts have been made.